src/main.rs

```diff
@@ -162,14 +152,16 @@ fn main() {
 				let decrypted = env_closure.db.decrypt_database(password.get());
 				match decrypted {
 					Ok(()) => {
-						error.set(String::from(""));
-						password.update(|pass| pass.zeroize());
-						app_state.set(AppState::Ready);
+						untrack(|| {
+							password.update(|pass| pass.zeroize());
+							toast_signals.kill_all_toasts();
+							app_state.set(AppState::Ready);
+						});
 					},
 					Err(err) => {
-						eprintln!("{:#?}", err);
-						error.set(err.to_string());
-						app_state.set(AppState::Ready);
+						untrack(|| {
+							toast_signals.add(err.to_string());
+						});
```

Comment:
This took me ages to figure out. Without `untrack` this lands us in an infinite loop. So I also added it to the `Ok` branch as I felt the signals feel a bit brittle. Now we're explicit that we don't want to run the effect again on changes inside of it.